src/assets/githubAction/githubActionBeginner.md

```diff
@@ -0,0 +1,130 @@
+
+Note:
+One event can trigger many workflows, a workflow can contain many jobs, and a job can contain many steps.
+
+### Example:
+
+![process_in_githubActions](/src/assets/githubAction/images/process.png)
+
+## Getting started:
+
+Before we dive into an example, make sure you have a repository where you want to set up your CI/CD workflow. If you don't have one, create a new repository for example named `action-hero`.
+
+Step 1: Clone your repository into your system
+```bash
+git clone <your_github_repo_url>
+```
+
+Step 2: Go to path `action-hero` and make a folder `.github`
+```bash
+cd action-hero
+mkdir .github
+```
+
+Step 4: Then, go to path `.github` and make a folder `workflows`
+```bash
+cd .github
+mkdir workflows
+```
+
+Step 5: Inside the folder `workflows` create a `hello_world.yml` file
+```bash
+cd workflows
+touch hello_world.yml
+```
+Our file structure will look like this
+```
+📦Action-hero
+┗ 📂.github
+┃ ┣ 📂workflows
+┃ ┃ ┗ 📜hello_world.yml
+```
+
+Step 6: Add the following code snippet to your `hello_world.yml` file:
+
+```
+name: Hello world workflow
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Print a message
+      run: echo "Hello, GitHub Actions!"
+```
+
+In this example:
+- The workflow is triggered by a push event on the main branch.
+- The job named "build" runs on the latest version of Ubuntu.
+- The first step checks out the repository using the actions/checkout action.
+- The second step simply prints a message using the echo command.
+
+## GitHub Actions activity:
+
+![github activity](/src/assets/githubAction/images/output.png)
+
+Above image is an output from a GitHub Actions workflow run. The output shows that the workflow was successful and completed the `Hello world workflow` job, which consisted of two steps: `Checkout repository` and `Print a message`.
+
+The `Print a message` step executed the command `echo "Hello, GitHub Actions!"`, resulting in the output `Hello, GitHub Actions!` being displayed in the workflow log.
+
+## Conclusion:
+In this blog, we've explored GitHub Actions, an automation and CI/CD service provided by GitHub. We've learned about its key components and how they work together to automate software development workflows.
+
+GitHub Actions enables us to automate tasks like building, testing, and deploying code, leading to improved software quality and time savings.
+
+With a practical example, we've seen how easy it is to set up a basic workflow.
+
+In summary, GitHub Actions is a powerful tool for speedy development processes, making it easier to deliver high-quality software efficiently.
```

Comment:
```suggestion
In summary, GitHub Actions is a powerful tool for speedy development processes, making it easier to deliver high-quality of software efficiently.
```

Reply:
@SagarGi if "high-quality" is used as an attribute to specify the type/matter of software the preposition "of" is not needed. Alternatively you could say software of high-quality, but I think this expression is less common. Maybe ask @phil-davis or your English teacher :)

Reply:
Correct. "high-quality software" is good.
"high-quality" is the adjective (describes an attribute of the software)
"software" is the noun (thing)